src/wellKnownDidConfiguration/index.ts

```diff
@@ -61,155 +51,269 @@ function checkOrigin(input: string) {
 ): Promise<DidUri> {
-  const { issuer, credentialSubject, proof } = credential
-
-  if (issuer !== credentialSubject.id) throw new Error('issuer and credential subject must be identical')
+  checkIsDomainLinkageCredential(credential)
 
-  const didUri = credentialSubject.id
-  Did.validateUri(didUri, 'Did')
+  const { credentialSubject, proof } = credential
+  const did = credentialSubject.id
 
   if (expectedOrigin !== credentialSubject.origin) throw new Error('origin does not match expected')
-  if (expectedDid && expectedDid !== didUri) throw new Error('DID does not match expected')
+  if (expectedDid && expectedDid !== did) throw new Error('DID does not match expected')
 
-  // get root hash incl fallback for older domain linkage credential types
-  const { rootHash = proof.rootHash, ...cleanSubject } = credentialSubject as any
-  if (!isHex(rootHash)) {
-    throw new Error('rootHash must be a hex encoded string')
+  const now = new Date().getTime()
+  if (new Date(credential.issuanceDate).getTime() > now) {
+    throw new Error('issuanceDate is in the future')
   }
-
-  const pType = Array.isArray(proof.type) ? proof.type : [proof.type]
-  if (!pType.includes(KILT_SELF_SIGNED_PROOF_TYPE)) {
-    throw new Error(`proof type must include ${KILT_SELF_SIGNED_PROOF_TYPE}`)
+  if (new Date(credential.expirationDate).getTime() < now) {
+    throw new Error('expirationDate is in the past')
   }
 
-  await Did.verifyDidSignature({
-    expectedVerificationMethod: 'assertionMethod',
-    signature: hexToU8a(proof.signature),
-    keyUri: proof.verificationMethod as DidResourceUri,
-    message: Utils.Crypto.coToUInt8(rootHash),
-  })
-
-  if (pType.includes(KILT_CREDENTIAL_DIGEST_PROOF_TYPE)) {
-    await verification.verifyCredentialDigestProof(
-      {
-        ...credential,
-        credentialSubject: cleanSubject,
-        id: `kilt:cred:${rootHash}`,
-      } as unknown as VerifiableCredential,
-      { ...proof, type: KILT_CREDENTIAL_DIGEST_PROOF_TYPE }
-    )
+  switch (proof.type) {
+    case KILT_SELF_SIGNED_PROOF_TYPE: {
+      // @ts-expect-error rootHash is a fallback for older domain linkage credential types
+      const { rootHash } = credentialSubject
+      let docHash: Uint8Array
+      if (rootHash && allowUnsafe) {
+        docHash = hexToU8a(rootHash)
+      } else {
+        const copy = JSON.parse(JSON.stringify(credential))
+        delete copy.credentialSubject.rootHash
+        delete copy.proof.signature
+        docHash = blake2AsU8a(JSON.stringify(copy))
+      }
+      await Did.verifyDidSignature({
+        expectedVerificationMethod: 'assertionMethod',
+        signature: hexToU8a(proof.signature),
+        keyUri: proof.verificationMethod as DidResourceUri,
+        message: docHash,
+      })
+      break
+    }
+    case DATA_INTEGRITY_PROOF_TYPE: {
+      const cryptosuite = suites.find(({ name }) => name === proof.cryptosuite)
```

Comment:
Shouldn't this be cryptoSuite?

Reply:
they are called `cryptosuite` in all lower case in the specs, so I'd say this is fine.